### PR TITLE
ADIN1110 and networking updates

### PR DESCRIPTION
**Pull request by CiprianRegus**
## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [x] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/net/adin1110/adin1110.c [hunk 1]
```diff
@@ -865,57 +865,63 @@ int adin1110_init(struct adin1110_desc **desc,
...
-		ADIN1110_MAC_LEN);
+	if (descriptor->reset_gpio) {
+		ret = no_os_gpio_direction_output(descriptor->reset_gpio, NO_OS_GPIO_LOW);
+		if (ret)
+			goto free_rst_gpio;
 
-	ret = no_os_gpio_get(&descriptor->reset_gpio, &param->reset_param);
-	if (ret)
-		goto free_spi;
+		no_os_mdelay(10);
```
Comment (amiclaus) [inline drivers/net/adin1110/adin1110.c:877]: some datasheet related comments would be nice for these specific values.

File: drivers/net/adin1110/adin1110.c [hunk 2]
```diff
@@ -865,57 +865,63 @@ int adin1110_init(struct adin1110_desc **desc,
...
+
+		ret = no_os_gpio_set_value(descriptor->reset_gpio, NO_OS_GPIO_HIGH);
+		if (ret)
+			goto free_rst_gpio;
 
-	ret = no_os_gpio_direction_output(descriptor->reset_gpio, NO_OS_GPIO_HIGH);
+		no_os_mdelay(90);
```
Comment (amiclaus) [inline drivers/net/adin1110/adin1110.c:883]: ditto

**Review by amiclaus — COMMENTED** [review]
looks good. minor comments.

**Comment by CiprianRegus** [comment]
- Added comments for the reset delay timing.
- Added 2 other fixes related to the lwip sockets implementation

**Review by amiclaus — APPROVED** [review]

**Review by buha — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]

**Pull request merged**